core/msg.c

```diff
@@ -370,6 +370,25 @@ static int _msg_receive(msg_t *m, int block)
     DEBUG("This should have never been reached!\n");
 }
 
+int msg_avail(void)
+{
+    unsigned state = disableIRQ();
+    DEBUG("msg_available: %" PRIkernel_pid ": msg_available.\n",
+          sched_active_thread->pid);
+
+    tcb_t *me = (tcb_t*) sched_threads[sched_active_pid];
```

Comment:
The `sched_threads[sched_active_pid]` changes should go in a separate PR. The reviewer needs to look up all write occurences of sched_active_thread and sched_active_pid to ensure that we are not introducing a bug (possibly a race while switching threads) in the message passing system.